### PR TITLE
Add Authentication Support

### PR DESCRIPTION
**Pull request by lverns**
This is most of the code for #2.

There still remains testing and UX/UI stuff, but it is shippable.

### CODE REVIEW

**Comment by lverns** [comment]
Merged master in to resolve conflicts, but there are a couple tweaks to make before this should be merged.

**Comment by lverns** [comment]
@frogrammer I've added some documentation to this and moved the hard-coded values out into the config file, so this is ready to merge unless you have concerns. I'll post the Google API keys to Slack.

**Comment by lverns** [comment]
Oh wait, I forgot to put everybody else in the admin list.

**Comment by frogrammer** [comment]
@lverns I still see some hard-coded `localhost` in a few places but we can change them later?

File: server/src/main/java/umm3601/digitalDisplayGarden/Authentication/Auth.java [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        this.clientSecret = clientSecret;
+        this.gson = new Gson();
+        this.authUsers = new ArrayList<>();
+        authUsers.add("gordo580@morris.umn.edu");
+        authUsers.add("schr1230@morris.umn.edu");
+        this.globalService = new ServiceBuilder()
+                .apiKey(clientId)
+                .apiSecret(clientSecret)
+                .scope("email") // replace with desired scope
+                .callback("http://localhost:2538/callback")
```
Comment (frogrammer) [inline server/src/main/java/umm3601/digitalDisplayGarden/Authentication/Auth.java]: This is one example of hard-code localhost.

**Comment by lverns** [comment]
Yeah, I just noticed that and am fixing that right now.

**Comment by lverns** [comment]
okay, I think I got them all

**Comment by frogrammer** [comment]
Seems good. Let me merge it.

**Pull request merged**